### PR TITLE
RSDK-5179 Numato bug fixes 

### PR DESCRIPTION
**Pull request by oliviamiller**
Verified new analog config works on the numato, fixed issues with closing cleanly. Thank you Alan for your help debugging this!

### CODE REVIEW

File: components/board/analog_smoother.go
Comment (randhid) [inline components/board/analog_smoother.go:123]: Please specify this if the analog smoother is connected through and reading serial, not through a spi bus as the pi/mcp3xxx does. 

Also - let's double-check that this doesn't cause any analog reader bug with the pi/jetson and one of the mcp3008 chips once this is merged. You can make a follow-up ticket.

File: components/board/numato/board.go
Comment (randhid) [inline components/board/numato/board.go:360]: nice workaround to an annoying issue.

**Review by randhid — APPROVED** [review]
LGTM! One nit and a follow-up ticket once this is merged, but not blocking on that.

File: components/board/numato/board.go
Comment (penguinland) [inline components/board/numato/board.go:359]: I think we need a much more detailed comment here. It took us hours to figure this out, and the nuance is not contained in this line. It should be something more like the following:

> The coroutine is blocked in the call to `in.ReadString`, and there is no way to interrupt that by cancelling a context or other signals. We also cannot move the `in.ReadString` call into `doSendReceive` directly, because then there is no way for it to time out if the board is not set up properly. Instead, we need the coroutine to complete that call so it wakes up and sees it should exit. Although closing the device port will accomplish this on certain OSes, on Mac you cannot close a device that is currently being read from, resulting in a deadlock (we can't close it until the coroutine stops reading, and the coroutine won't stop reading if it's not closed). So instead, we send the Numato board a command. The response back will wake up the coroutine, which will then notice it's time to exit, and we can close the device afterwards.

(that was too wordy. There's probably a good compromise in between your version and mine)

File: components/board/numato/board.go
Comment (penguinland) [inline components/board/numato/board.go:431]: This totally works. Another way that keeps the code less indented is to make this part
```go
b.activeBackgroundWorkers.Add(1)
utils.ManagedGo(b.readThread, b.activeBackgroundWorkers.Done)
```
and then leave the implementation of `readThread` largely unchanged. I'm content with either approach, though I think this suggestion is easier to reason about because `readThread` itself can just focus on reading the thread, and the background workers are only mentioned in the constructor and destructor (`connect` and `Close`).

**Review by penguinland — APPROVED** [review]
All of my feedback is optional. LGTM.

File: components/board/analog_smoother.go
Comment (randhid) [reply components/board/analog_smoother.go:120]: The serial port refers to communications protocol - USB is one of the users of serial, in this day and age - it could refer to a usbA looking thing that speaks over RS-232 or RS-485, or a canbut looking thing that speaks over those two protocols as well.

File: components/board/analog_smoother.go
Comment (oliviamiller) [reply components/board/analog_smoother.go]: Found a much cleaner solution to just close the analog smoother in the numato close function. Now we don't have to worry about this breaking analogSmoother for boards using a MCP300x ADC.

Comment (penguinland) [reply components/board/analog_smoother.go]: Yes, I like the new version much more!

**Pull request merged**
